lib/wasmex/components/component.ex

```diff
@@ -31,4 +31,40 @@ defmodule Wasmex.Components.Component do
       resource -> {:ok, __wrap_resource__(resource)}
     end
   end
+
+  defmacro __using__(opts) do
+    genserver_stuff =
+      quote do
+        use GenServer
+
+        def start_link(opts) do
+          Wasmex.Components.start_link(opts)
+        end
+
+        def handle_call(request, from, state) do
+          Wasmex.Components.handle_call(request, from, state)
+        end
+      end
+
+    functions =
+      if wit_path = Keyword.get(opts, :wit) do
+        wit_contents = File.read!(wit_path)
+        exported_functions = Wasmex.Native.wit_exported_functions(wit_path, wit_contents)
+
+        for {function, arity} <- exported_functions do
```

Comment:
can't wait for imported function support! :D

Reply:
yes, definitely high on my list of next things.